src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -527,20 +527,31 @@ def to_analysis_target(config: Configuration, available_domains: list[str]) -> A
         InvalidPURLError
             If the PURL provided from the user is invalid.
         """
-        # Due to the current design of Configuration class. purl, repo_path, branch and digest are initialized
-        # as empty strings and we assumed that they are always set with input values as non-empty strings.
+        # Due to the current design of Configuration class, repo_path, branch and digest are initialized
+        # as empty strings, and we assumed that they are always set with input values as non-empty strings.
         # Therefore, their true types are ``str``, and an empty string indicates that the input value is not provided.
-        purl_input_str: str = config.get_value("purl")
+        # The purl might be a PackageURL type, or a string, and therefore requires extra care.
+        parsed_purl: PackageURL | None
+        if config.get_value("purl") is None or config.get_value("purl") == "":
+            parsed_purl = None
+        elif isinstance(config.get_value("purl"), str):
+            try:
+                parsed_purl = PackageURL.from_string(config.get_value("purl"))
+            except InvalidPURLError as error:
+                raise PURLNotFoundError("Invalid input PURL.") from error
```

Comment:
I think `InvalidPURLError` would be more suitable to be raised. Also, it would be helpful to include the `config.get_value("purl"),` in the error.